ghost/email-service/lib/mailgun-email-provider.js

```diff
@@ -144,16 +144,29 @@ class MailgunEmailProvider {
             return {
                 id: response.id.trim().replace(/^<|>$/g, '')
             };
-        } catch ({error, messageData}) {
-            // REF: possible mailgun errors https://documentation.mailgun.com/en/latest/api-intro.html#status-codes
-            let ghostError = new errors.EmailError({
-                statusCode: error.status,
-                message: this.#createMailgunErrorMessage(error),
-                errorDetails: JSON.stringify({error, messageData}),
-                context: `Mailgun Error ${error.status}: ${error.details}`,
-                help: `https://ghost.org/docs/newsletters/#bulk-email-configuration`,
-                code: 'BULK_EMAIL_SEND_FAILED'
-            });
+        } catch (e) {
+            let ghostError;
+            if (e.error && e.messageData) {
+                const {error, messageData} = e;
+                
+                // REF: possible mailgun errors https://documentation.mailgun.com/en/latest/api-intro.html#status-codes
+                ghostError = new errors.EmailError({
+                    statusCode: error.status,
+                    message: this.#createMailgunErrorMessage(error),
+                    errorDetails: JSON.stringify({error, messageData}),
+                    context: `Mailgun Error ${error.status}: ${error.details}`,
+                    help: `https://ghost.org/docs/newsletters/#bulk-email-configuration`,
+                    code: 'BULK_EMAIL_SEND_FAILED'
+                });
+            } else {
+                ghostError = new errors.EmailError({
+                    statusCode: undefined,
+                    message: e.message,
+                    errorDetails: undefined,
+                    context: e.context || 'Mailgun Error',
+                    code: 'BULK_EMAIL_SEND_FAILED'
+                });
+            }
```

Comment:
We need to catch code errors apart from Mailgun errors, so this change was needed to prevent that the catch would also throw